astroquery/mast/cloud.py

```diff
@@ -0,0 +1,220 @@
+            log.warning("Your AWS account will be charged for access to the S3 bucket")
+            log.info("See Request Pricing in https://aws.amazon.com/s3/pricing/ for details")
+            log.info("If you have not configured boto3, follow the instructions here: "
+                     "https://boto3.readthedocs.io/en/latest/guide/configuration.html")
+
+    def is_supported(data_product):
+        """
+        Given a data product, determines if it is in a mission available in the cloud.
+
+        Parameters
+        ----------
+        data_product : `~astropy.table.Row`
+            Product to be validated.
+
+        Returns
+        -------
+        response : bool
+              Is the product from a supported mission.
+        """
+
+        for mission in self.supported_missions:
+            if data_product['dataURI'].lower().startswith(mission):
+                return True
+        return False
+
+    def get_cloud_uri(self, data_product, include_bucket=True, full_url=False):
+        """
+        For a given data product, returns the associated cloud URI.
+        If the product is from a mission that does not support cloud access an
+        exception is raised. If the mission is supported but the product
+        cannot be found in the cloud, the returned path is None.
+
+        Parameters
+        ----------
+        data_product : `~astropy.table.Row`
+            Product to be converted into cloud data uri.
+        include_bucket : bool
+            Default True. When false returns the path of the file relative to the
+            top level cloud storage location.
+            Must be set to False when using the full_url argument.
+        full_url : bool
+            Default False. Return an HTTP fetchable url instead of a cloud uri.
+            Must set include_bucket to False to use this option.
+
+        Returns
+        -------
+        response : str or None
+            Cloud URI generated from the data product. If the product cannot be
+            found in the cloud, None is returned.
+        """
+
+        s3_client = self.boto3.client('s3')
+
+        path = utils.mast_relative_path(data_product["dataURI"])
+        if path is None:
+            raise InvalidQueryError("Malformed data uri {}".format(data_product['dataURI']))
+
+        path = path.lstrip("/")
+
+        try:
+            s3_client.head_object(Bucket=self.pubdata_bucket, Key=path, RequestPayer='requester')
+            if include_bucket:
+                path = "s3://{}/{}".format(self.pubdata_bucket, path)
+            elif full_url:
+                path = "http://s3.amazonaws.com/{}/{}".format(self._pubdata_bucket, path)
+            return path
+        except self.botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] != "404":
+                raise
+
+        warnings.warn("Unable to locate file {}.".format(data_product['productFilename']), NoResultsWarning)
+        return None
```

Comment:
this line is redudant; `return` would do the same thing, as would leaving this line out.

Reply:
True. I was following the pep8 guidance about return statement consistency:
"Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None, and an explicit return statement should be present at the end of the function (if reachable)"

Reply:
Oh, ok - fair enough!  I didn't know about this pep8 guideline.